server/commands/check.go

```diff
@@ -110,16 +108,40 @@ func (query *CheckQuery) Execute(ctx context.Context, req *openfgapb.CheckReques
 }
 
 // getTypeDefinitionRelationUsersets validates a tuple and returns the userset corresponding to the "object" and "relation"
-func (query *CheckQuery) getTypeDefinitionRelationUsersets(ctx context.Context, rc *resolutionContext) (*openfgapb.Userset, error) {
-	ctx, span := query.tracer.Start(ctx, "getTypeDefinitionRelationUsersets")
-	defer span.End()
+func getTypeDefinitionRelationUsersets(tk *openfgapb.TupleKey, typesys *typesystem.TypeSystem) (*openfgapb.Userset, error) {
+
+	objectType := tupleUtils.GetType(tk.GetObject())
+
+	relation, err := typesys.GetRelation(objectType, tk.GetRelation())
```

Comment:
Can we move these two lines below the `ValidateTuple` call? And then we can remove the error handling code here because that method would have thrown

Reply:
We can, but we still need an error handling block. Whether we panic or handle the error based on the return contract is up to us, but we shouldn't just throw away the error. I'm wondering if we should log errors here instead 🤔 ..? But then any "invalid" tuples because of model differences would be logged and that would suck.. 🤮 

Not sure the best way to handle this in a way that feels elegant at all.

Reply:
Future PR!